builds/jenkins-build-pack-api/Jenkinsfile

```diff
@@ -28,511 +29,521 @@ import groovy.transform.Field
+        environmentDeploymentMetadata.setEnvironmentEndpoint(endpointUrl)
+        events.sendStartedEvent('UPDATE_ENVIRONMENT', 'Environment status update event for staging deployment', environmentDeploymentMetadata.generateEnvironmentMap("deployment_started", environment_logical_id), environment_logical_id)
+        withCredentials([
+          [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']
+        ]) {
+          echo "AWS Configure ......."
+          sh "aws configure set profile.cloud-api.region ${configLoader.AWS.REGION}"
+          sh "aws configure set profile.cloud-api.aws_access_key_id $AWS_ACCESS_KEY_ID"
+          sh "aws configure set profile.cloud-api.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+
+          // Generate serverless yml file with domain added in function name
+          echo "Generate deployment env with domain for ${env_key}"
+
+          writeServerlessFile(config, environment_logical_id)
+
+          setLambdaExecutionRole(configLoader.AWS.ROLEID);
+
+          echo "serverless deploy......."
+          def envBucketKey = "${env_key}${configLoader.JAZZ.S3_BUCKET_NAME_SUFFIX}"
+          sh "serverless deploy --stage ${environment_logical_id} -v --bucket ${configLoader.AWS.S3[envBucketKey]}"
+
+          def lambdaARN = getLambdaARN(stackName);
+          events.sendCompletedEvent('CREATE_ASSET', null, utilModule.generateAssetMap("aws", lambdaARN, "lambda", config), environment_logical_id);
+
+          if (fileExists('swagger/swagger.json')) {
+            echo "Generating swagger file for environment: ${env_key}"
+
+            def api_id = utilModule.getAPIId(configLoader.AWS.API[env_key], config)
+            echo "API ID: $api_id"
+
+            def apiHostName = "${api_id}.execute-api.${configLoader.AWS.REGION}.amazonaws.com"
+            echo "API Host Name: $apiHostName"
+            generateSwaggerEnv(environment_logical_id, "${configLoader.INSTANCE_PREFIX}-${current_environment}", apiHostName, config)
+            if (env_key.equals("DEV") && config['domain'] != 'jazz') {
+              updateStageForDev(environment_logical_id)
+            }
+
+            sh "cat swagger/swagger.json "
+
+            echo "Deploying to API Gateway environment.."
+            sh "aws apigateway put-rest-api --rest-api-id ${api_id} --mode merge --parameters basepath=prepend --body 'file://swagger/swagger.json'" + " --profile cloud-api"
+            sh "aws apigateway create-deployment --rest-api-id ${api_id} --stage-name ${current_environment} --profile cloud-api"
+            sh "aws apigateway tag-resource --resource-arn arn:aws:apigateway:${configLoader.AWS.REGION}::/restapis/${api_id}/stages/${current_environment} --tags Application=Jazz,JazzInstance=${configLoader.INSTANCE_PREFIX}  --profile cloud-api"
+
+            if ("${configLoader.JAZZ.API_DETAILED_MONITORING}" == "true") {
```

Comment:
We will use the value of that config and set it directly. This way when its turned off too, it will work as expected.

Reply:
Done